src/googmodule.ts

```diff
@@ -0,0 +1,388 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as path from 'path';
+
+import {ModulesManifest} from './modules_manifest';
+import {visitNodeWithSynthesizedComments} from './transformer_util';
+import * as ts from './typescript';
+
+export interface GoogModuleProcessorHost {
+  /**
+   * Takes a context (the current file) and the path of the file to import
+   *  and generates a googmodule module name
+   */
+  pathToModuleName(context: string, importPath: string): string;
+  /**
+   * If we do googmodule processing, we polyfill module.id, since that's
+   * part of ES6 modules.  This function determines what the module.id will be
+   * for each file.
+   */
+  fileNameToModuleId(fileName: string): string;
+  /** Whether to convert CommonJS module syntax to `goog.module` Closure imports. */
+  googmodule?: boolean;
```

Comment:
Why does the GoogModuleProcessor have a googmodule boolean?  What does it mean for this to be false?

Reply:
Actually, it turns out with this code change we catch whether to do the `googmodule` transformation on a higher level, so this can indeed go away.